lib/utils/link.js

```diff
@@ -25,6 +25,24 @@ function linkIfExists (from, to, gently, cb) {
   })
 }
 
+function resolveIfSymlink (maybeSymlinkPath, cb) {
+  fs.lstat(path.resolve(maybeSymlinkPath), function (err, stat) {
+    if (err) return cb.apply(this, arguments)
+    if (!stat.isSymbolicLink()) return cb(null, maybeSymlinkPath)
+    fs.readlink(path.resolve(maybeSymlinkPath), cb)
+  })
+}
+
+function ensureFromIsNotSource (from, to, cb) {
+  resolveIfSymlink(from, function (err, from) {
+    if (err) return cb.apply(this, arguments)
+    if (from === path.resolve(to)) {
```

Comment:
`from` here is the destination of the symlink, which might be relative and so probably _should_ be passed through `path.resolve` too, but it should be relative to where the symlink was, not cwd. So something like `path.resolve(path.dirname(from), fromDestination)`.